src/lib/lspace/elements/root_element.rs

```diff
@@ -8,91 +11,119 @@ use elements::element::{TElement, ElementRef};
 use elements::container::TContainerElement;
 
 
-pub struct RootElement {
+struct RootElementMut {
     req: ElementReq,
     alloc: ElementAlloc,
     children: Vec<ElementRef>,
 }
 
+pub struct RootElement {
+    m: RefCell<RootElementMut>,
+}
+
 impl RootElement {
     pub fn new(child: ElementRef) -> RootElement {
-        return RootElement{req: ElementReq::new(), alloc: ElementAlloc::new(),
-                           children: vec![child]};
+        return RootElement{m: RefCell::new(RootElementMut{
+                                req: ElementReq::new(), alloc: ElementAlloc::new(),
+                                children: vec![child]})};
     }
 
-    pub fn root_requisition_x(&mut self) -> f64 {
+    pub fn root_requisition_x(&self) -> f64 {
         self.update_x_req();
-        return self.req.x_req.size().size();
+        return self.m.borrow().req.x_req.size().size();
     }
 
-    pub fn root_allocate_x(&mut self, width: f64) {
-        self.alloc.x_alloc = LAlloc::new_from_req_in_avail_size(&self.req.x_req, 0.0, width);
+    pub fn root_allocate_x(&self, width: f64) {
+        // The following line of code would make sense in other languages:
+        //
+        //    self.m.borrow_mut().alloc.x_alloc =
+        //          LAlloc::new_from_req_in_avail_size(&self.m.borrow().req.x_req, 0.0, width);
+        //
+        // The mutable borrow for the assignment could only be taken out after the value has been computed;
```

Comment:
These lines are way too long (100 chars is a reasonable max, though some people prefer 80).